Microsoft.Toolkit.Uwp.UI/Triggers/ControlSizeTrigger.cs

```diff
@@ -0,0 +1,170 @@
+            typeof(ControlSizeTrigger),
+            new PropertyMetadata(double.PositiveInfinity, (d, e) => ((ControlSizeTrigger)d).UpdateTrigger()));
+
+        /// <summary>
+        /// Gets or sets the min height at which to trigger.
+        /// </summary>
+        public double MinHeight
+        {
+            get => (double)GetValue(MinHeightProperty);
+            set => SetValue(MinHeightProperty, value);
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="MinHeight"/> DependencyProperty.
+        /// </summary>
+        public static readonly DependencyProperty MinHeightProperty = DependencyProperty.Register(
+            nameof(MinHeight),
+            typeof(double),
+            typeof(ControlSizeTrigger),
+            new PropertyMetadata(0.0, (d, e) => ((ControlSizeTrigger)d).UpdateTrigger()));
+
+        /// <summary>
+        /// Gets or sets the element whose width will observed
+        /// for the trigger.
+        /// </summary>
+        public FrameworkElement TargetElement
+        {
+            get => (FrameworkElement)GetValue(TargetElementProperty);
+            set => SetValue(TargetElementProperty, value);
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="TargetElement"/> DependencyProperty.
+        /// </summary>
+        /// <remarks>
+        /// Using a DependencyProperty as the backing store for TargetElement. This enables animation, styling, binding, etc.
+        /// </remarks>
+        public static readonly DependencyProperty TargetElementProperty = DependencyProperty.Register(
+            nameof(TargetElement),
+            typeof(FrameworkElement),
+            typeof(ControlSizeTrigger),
+            new PropertyMetadata(null, OnTargetElementPropertyChanged));
+
+        private static void OnTargetElementPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            ((ControlSizeTrigger)d).UpdateTargetElement((FrameworkElement)e.OldValue, (FrameworkElement)e.NewValue);
+        }
+
+        // Handle event to get current values
+        private void OnTargetElementSizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            UpdateTrigger();
+        }
+
+        private void UpdateTargetElement(FrameworkElement oldValue, FrameworkElement newValue)
+        {
+            if (oldValue != null)
+            {
+                oldValue.SizeChanged -= OnTargetElementSizeChanged;
+            }
+
+            if (newValue != null)
+            {
+                newValue.SizeChanged += OnTargetElementSizeChanged;
+            }
+
+            UpdateTrigger();
+        }
+
+        // Logic to evaluate and apply trigger value
+        private void UpdateTrigger()
+        {
+            if (TargetElement == null || !CanTrigger)
+            {
+                SetActive(false);
+                return;
+            }
+
+            SetActive(
+                MinWidth <= TargetElement.ActualWidth &&
```

Comment:
We probably want to clarify in the comments for the min/max properties whether the bound is inclusive or exclusive of the value. I think you've done this so that you can define ranges like Min=300 to Max=400, and then another trigger for Min=400 to Max=500 and not have overlap, right?

I like the Min inclusive as that's how AdaptiveTrigger I think works today.